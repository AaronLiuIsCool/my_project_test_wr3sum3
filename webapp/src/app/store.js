import { configureStore } from '@reduxjs/toolkit';
import resumeReducer from 'features/SmartResume/slicers/resumeSlicer';
import counterReducer from 'features/CounterExample/counterSlice.js';

export default configureStore({
  reducer: {
    counter: counterReducer,
    resume: resumeReducer
  },
});
