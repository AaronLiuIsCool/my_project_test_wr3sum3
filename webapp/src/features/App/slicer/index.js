import { createSlice } from '@reduxjs/toolkit';

const appSlicers = createSlice({
  name: 'app',
  initialState: {
    language: 'zh'
  },
  reducers: {
    updateLanguage: (state, action) => {
      state.language = action.payload;
    },
  },
});

export const { updateLanguage } = appSlicers.actions;

export const selectLanguage = ({app}) => app.language;

export default appSlicers.reducer;
