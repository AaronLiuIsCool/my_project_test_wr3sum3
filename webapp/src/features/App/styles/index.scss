.App {
  height: 100%;

  nav {
    height: 80px;
    border-bottom: 1px solid #f2f4f9;

    .navbar-brand {
      margin: 0 10px;
    }

    .nav-link {
      margin-left: 2.5em !important;
      font-size: 1rem !important;
      color: #bfc4d3 !important;
      line-height: 64px;

      &.active {
        color: #333333 !important;
        font-weight: bold;
        border-bottom: 2px solid #3e89ec;
      }
    }
  }

  .features {
    &.padding-for-nav {
      padding-top: 80px;
    }
  }
}
