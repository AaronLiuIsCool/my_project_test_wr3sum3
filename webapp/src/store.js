import { configureStore } from '@reduxjs/toolkit';

import appReducer from 'features/App/slicer';
import resumeReducer from 'features/SmartResume/slicer';

export default configureStore({
  reducer: {
    app: appReducer,
    resume: resumeReducer
  },
});
