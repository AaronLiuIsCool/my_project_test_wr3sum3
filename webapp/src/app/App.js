import React, { useEffect, useState } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { useSelector } from 'react-redux';

import { I8nContext } from 'shell/i18n';
import { selectLanguage } from './slicer';

import Navigation from './components/Navigation';
import SmartResume from 'features/SmartResume';
import Counter from 'features/CounterExample';

import './styles/App.css';

import zh from './i18n/zh.json';
import en from './i18n/en.json';

const App = () => {
  const language = useSelector(selectLanguage);
  const [messages, setMessages] = useState({});

  useEffect(() => {
    setMessages(language === 'zh' ? zh : en);
  }, [language]);

  return (
    <I8nContext.Provider value={messages}>
      <BrowserRouter>
        <div className="App">
          <Navigation />
          <Route exact path="/" component={SmartResume} />
          <Route path="/example" component={Counter} />
        </div>
      </BrowserRouter>
    </I8nContext.Provider>
  );
};

export default App;
