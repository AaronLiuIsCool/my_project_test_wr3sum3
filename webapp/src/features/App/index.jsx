import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';

import moment from 'moment';
import 'moment/locale/zh-cn';

import { GApageView } from 'utils/GATracking';

import AuthServices from 'shell/services/AuthServices';
import { I8nContext } from 'shell/i18n';
import { getLogger } from 'shell/logger';
import { selectLanguage, updateAuthInfo } from './slicer';

import PrivateRoute from './components/PrivateRoute';
import Navigation from './components/Navigation';
import ResumeHub from 'features/ResumeHub';
import ResumeStarter from 'features/ResumeStarter';
import SmartResume from 'features/SmartResume';
import JobsMatcher from 'features/JobsMatcher';
import JobCollection from 'features/JobCollection';
import JobRefinement from 'features/JobRefinement';
import Page404 from 'features/SpecialPages/Page404';

import './styles/index.scss';

import zh from './i18n/zh.json';
import en from './i18n/en.json';

const authServices = new AuthServices();
const logger = getLogger('App');
async function isAuthenticated(dispatch) {
  let authInfo = {};
  try {
    authInfo = await authServices.findWhoAmI();
  } catch (exception) {
    logger.error(exception);
  } finally {
    dispatch(updateAuthInfo(authInfo));
  }
}
const App = ({ waitForInit = true }) => {
	const [init, setInit] = useState(false);
	const language = useSelector(selectLanguage);
	const dispatch = useDispatch();

  useEffect(() => {
    GApageView(); // call GA for page viewed
    isAuthenticated(dispatch).then(() => setInit(true));
    const interval = window.setInterval(() => isAuthenticated(dispatch), 60 * 60 * 1000);
    return () => clearInterval(interval);
  }, []); // eslint-disable-line

	let messages = en;
	moment.locale('en');
	if (language === 'zh') {
		messages = zh;
		moment.locale('zh-cn');
	}

	if (waitForInit && !init) {
		return <React.Fragment />;
	}

  return (
    <I8nContext.Provider value={messages}>
      <BrowserRouter>
        <div className="App">
          <Navigation />
          <Switch>
            <PrivateRoute exact path="/" component={ResumeHub} />
            <PrivateRoute path="/new" component={ResumeStarter} />
            <PrivateRoute exact path="/resume" component={SmartResume} />
            <PrivateRoute exact path="/resume/:resumeId" component={SmartResume} />
            <PrivateRoute path="/jobs" component={JobsMatcher} />
            <PrivateRoute path="/refinement" component={JobRefinement} />
            <PrivateRoute path="/collections" component={JobCollection} />
            <Route component={Page404} />
            <Route path="/404" component={Page404} />
          </Switch>
        </div>
      </BrowserRouter>
    </I8nContext.Provider>
  );
};

App.propTypes = {
	waitForInit: PropTypes.bool,
};

export default App;
