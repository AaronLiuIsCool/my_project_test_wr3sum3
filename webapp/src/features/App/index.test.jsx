import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from 'store';
import App from '.';

test('renders App', () => {
  const app = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  ).toJSON();

  expect(app).toMatchSnapshot();
});
