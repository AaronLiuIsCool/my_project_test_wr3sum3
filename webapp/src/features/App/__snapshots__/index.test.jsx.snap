// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light fixed-top"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      快刀
    </a>
    <button
      aria-controls="top-navigation-bar"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="top-navigation-bar"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="#"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            智能创建
          </a>
        </a>
        <a
          className="nav-link"
          data-rb-event-key="#"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <a
            href="/jobs"
            onClick={[Function]}
          >
            工作匹配
          </a>
        </a>
      </div>
      <div
        className="btn-group btn-group-toggle"
        role="group"
      >
        <label
          className="btn active btn-primary"
          disabled={false}
        >
          <input
            autoComplete="off"
            checked={true}
            disabled={false}
            name="languageSelect"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="zh"
          />
          中文
        </label>
        <label
          className="btn btn-primary"
          disabled={false}
        >
          <input
            autoComplete="off"
            checked={false}
            disabled={false}
            name="languageSelect"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="en"
          />
          English
        </label>
      </div>
    </div>
  </nav>
  <div>
    Loading...
  </div>
</div>
`;
